modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/SidebarPanel.js

```diff
@@ -14,26 +14,82 @@
 
 import ClayAlert from '@clayui/alert';
 import ClayLoadingIndicator from '@clayui/loading-indicator';
-import {useTimeout} from 'frontend-js-react-web';
+import {useIsMounted} from 'frontend-js-react-web';
 import {fetch} from 'frontend-js-web';
-import React, {useEffect, useImperativeHandle, useRef, useState} from 'react';
+import React, {useEffect, useImperativeHandle, useReducer, useRef} from 'react';
 
 import Sidebar from './components/Sidebar';
 
+const dataReducer = (state, action) => {
+	switch (action.type) {
+		case 'CLOSE_SIDEBAR':
+			return {
+				...state,
+				isOpen: false,
+			};
+
+		case 'LOAD_DATA':
+			return {
+				...state,
+				data: null,
+				error: null,
+				loading: true,
+			};
+
+		case 'OPEN_SIDEBAR':
+			return {
+				...state,
+				isOpen: true,
+			};
+
+		case 'SET_ERROR':
+			return {
+				...state,
+				data: null,
+				error: action.error,
+				loading: false,
+			};
+
+		case 'SET_HTML':
+			return {
+				...state,
+				data: {
+					html: action.html,
+				},
+				error: null,
+				loading: false,
+			};
+
+		case 'SET_JSON':
+			return {
+				...state,
+				data: action.data,
+				error: action.data?.error,
+				loading: false,
+			};
+
+		default:
+			return initialState;
+	}
+};
+
+const initialState = {
+	data: null,
+	error: null,
+	loading: false,
+	open: true,
+};
+
 const SidebarPanel = React.forwardRef(
 	({fetchURL, onClose, viewComponent: View}, ref) => {
 		const CurrentView = useRef(View);
 
-		const [error, setError] = useState();
-		const [isLoading, setIsLoading] = useState();
-		const [isOpen, setIsOpen] = useState(true);
-		const [resourceData, setResourceData] = useState();
+		const isMounted = useIsMounted();
 
-		const delay = useTimeout();
+		const [state, dispatch] = useReducer(dataReducer, initialState);
 
 		const getData = (fetchURL) => {
-			setIsLoading(true);
-			setResourceData(null);
+			safeDispatch({type: 'LOAD_DATA'});
 
 			fetch(fetchURL, {
 				method: 'GET',
```

Comment:
Hey @carloslancha I know it was here before but maybe it would be a great opportunity to use `useResource` here. Just a suggestion 😬!

It would be equivalent:
```js
const customFetcher = (url, options) =>
  fetch(url, options).then((response) => response.headers.get('content-type').includes('html') ? response.text() : response);

const {data, refetch} = useResource({
  fetch: customFetcher,
  fetchPolicy: 'cache-first', // If that helps
  link: fetchURL,
  onNetworkStatusChange: (networkStatus) => {...}
});
```

Reply:
Hey Matu! Thx for the suggestion! What are the benefits of using useResource instead fetch?

Reply:
@carloslancha the biggest benefits are cache if it is necessary according to the policy that you think is best, like searching the cache first and then validating or just searching the cache and the [retry mechanism with jitter](https://clayui.com/docs/components/data-provider.html#retry).

Reply:
Thx Matu! I'll go with fetch in this case but I'll keep `useResource` in mind for further uses 😄